src/main/java/com/stemcraft/feature/SMWaystones.java

```diff
@@ -227,7 +232,14 @@ private void updateWaystoneForLocation(Location location) throws SQLException {
     private void updateWaystone(List<Location> locations) throws SQLException {
         STEMCraft.runLater(5, () -> {
             try {
+                List<String> allowedDimensionStringList = SMConfig.main().getStringList("waystones.allowedDimensions");
```

Comment:
Cache this data `onEnabled` instead of calling every method (which could be a lot!). See https://github.com/STEMMechanics/STEMCraft-Core/blob/680d2c46b14a55f9288653c36b04b8bb6b6d2dee/src/main/java/com/stemcraft/feature/SMGraves.java#L27 for an example of how SMGraves does this

Reply:
only problem with this is that it won't update on config changes

Reply:
or does it reenable plugins on config reload?